### ISSUE TITLE
[suggestion] Draw a circle

### ISSUE THREAD

**Issue by Shohreh** [umap-project/umap#1811]
Hello,

I sometimes need to draw a circle around a location instead of just a polygon (besides, it'd be useful to draw a perfect rectangle).

I suggest adding that feature to Umap.

Cheers,

![image](https://github.com/umap-project/umap/assets/1541861/bc842adc-7094-4bbf-b985-13530cc7e72b)

**Comment by yohanboniface** [comment]
Same as #1698 ?

**Comment by Shohreh** [comment]
Yes, although simpler — not specifically around waypoints, and without bothering with overlapping circles.

**Comment by kyeastmood** [comment]
Hi, I would also be happy to have this feature.

**Comment by kyeastmood** [comment]
Maybe a circle-resembling polygon would be enough?
Like in Turf: https://turfjs.org/docs/api/circle
Polygon roundness can be controlled with the `options.steps` parameter.

<img width="1028" height="1366" alt="Image" src="https://github.com/user-attachments/assets/862d992e-25a4-4b1a-af7d-408da5995709" />

**Comment by kyeastmood** [comment]
This is a Turf-generated "circle" with radius of 1 kilometer and 64 steps (the default):

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/b66f4958-4bce-4c6e-b9d3-cc6c0ebcde0a" />
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/a34677fc-96e2-4b0c-8814-912df2b81a0a" />

In my opinion it looks good.

**Comment by yohanboniface** [comment]
Leaflet already has a Circle marker: https://leafletjs.com/reference.html#circle
But it does not translate immediately in geojson spec, which uMap uses for storing data. It's not a big deal (could be just a Point with a radius property), but yet to be done… ;)
